hadoop-ozone/tools/src/test/java/org/apache/hadoop/ozone/debug/datanode/container/TestChecksumSubcommand.java

```diff
@@ -0,0 +1,214 @@
+    command.setTreeFilePath(treeFile.getAbsolutePath());
+
+    RuntimeException exception = assertThrows(RuntimeException.class, command::call);
+
+    assertThat(err.toString(DEFAULT_ENCODING)).contains("Error reading tree file");
+    assertThat(exception.getMessage()).contains("Failed to read tree file");
+  }
+
+  @Test
+  void testChecksumCommandWithEmptyFile() throws Exception {
+    // Create an empty tree file
+    File treeFile = new File(tempDir.toFile(), "empty.tree");
+    ContainerProtos.ContainerChecksumInfo emptyInfo = ContainerProtos.ContainerChecksumInfo.newBuilder().build();
+    try (OutputStream fos = Files.newOutputStream(treeFile.toPath())) {
+      emptyInfo.writeTo(fos);
+    }
+
+    JsonNode actualJson = runChecksumCommand(treeFile);
+    
+    // Build expected JSON string for empty container
+    String expectedJson = String.format(
+        "[ {" +
+        "\"containerID\" : 0," +
+        "\"filePath\" : \"%s\"" +
+        "} ]", 
+        treeFile.getAbsolutePath().replace("\\", "\\\\")
+    );
+    ObjectMapper mapper = JsonUtils.getDefaultMapper();
+    JsonNode expectedJsonNode = mapper.readTree(expectedJson);
+    
+    // Compare JSON structures
+    assertEquals(expectedJsonNode, actualJson);
+  }
+
+  @Test
+  void testChecksumCommandWithComplexTree() throws Exception {
+    // Create a mock container data
+    KeyValueContainerData containerData = Mockito.mock(KeyValueContainerData.class);
+    Mockito.when(containerData.getContainerID()).thenReturn(CONTAINER_ID);
+    Mockito.when(containerData.getMetadataPath()).thenReturn(tempDir.toString());
+
+    // Build a more complex test tree with more blocks
+    ContainerMerkleTreeWriter tree = buildTestTree(config, 10); // 10 blocks instead of default 5
+    updateTreeProto(containerData, tree.toProto());
+
+    File treeFile = new File(tempDir.toFile(), CONTAINER_ID + ".tree");
+
+    JsonNode actualJson = runChecksumCommand(treeFile);
+
+    // Verify the structure and key fields for complex tree
+    assertThat(actualJson.isArray()).isTrue();
+    assertThat(actualJson.size()).isEqualTo(1);
+
+    JsonNode containerJson = actualJson.get(0);
+    assertThat(containerJson.get("containerID").asLong()).isEqualTo(CONTAINER_ID);
+    assertThat(containerJson.get("filePath").asText()).isEqualTo(treeFile.getAbsolutePath());
+    assertThat(containerJson.has("containerMerkleTree")).isTrue();
+
+    JsonNode merkleTree = containerJson.get("containerMerkleTree");
+    assertThat(merkleTree.has("dataChecksum")).isTrue();
+    assertThat(merkleTree.has("blockMerkleTrees")).isTrue();
+    assertThat(merkleTree.get("blockMerkleTrees").isArray()).isTrue();
+    assertThat(merkleTree.get("blockMerkleTrees").size()).isEqualTo(10); // Complex tree has 10 blocks
+  }
+
+  /**
+   * Helper method to run ChecksumSubcommand and return parsed JSON output.
+   */
+  private JsonNode runChecksumCommand(File treeFile) throws Exception {
```

Comment:
I think we can assert that `stderr` is empty in this method too based on how it is used by the callers.